workers/proof-manager/src/mock.rs

```diff
@@ -31,134 +43,121 @@ pub struct MockProofManager;
 impl MockProofManager {
     /// Start a mock proof manager
     pub fn start(job_queue: Receiver<ProofManagerJob>) {
-        Handle::current().spawn_blocking(move || Self::execution_loop(&job_queue));
+        Handle::current().spawn_blocking(move || {
+            if let Err(e) = Self::execution_loop(&job_queue) {
+                log::error!("error in mock proof manager: {e}");
+            }
+        });
     }
 
     /// The execution loop for the mock
-    fn execution_loop(job_queue: &Receiver<ProofManagerJob>) {
+    fn execution_loop(job_queue: &Receiver<ProofManagerJob>) -> Result<(), ProofManagerError> {
         loop {
             match job_queue.recv() {
                 Err(e) => {
                     log::error!("error receiving job: {e}... shutting down");
```

Comment:
should we still log this? seems like the error will necessarily get logged above in `start`. and if we do wanna log here, should we still say "shutting down"?